File: 01_Language/01_Functions/js/explode.js
function explode(delimiter, string) {
    let res = []
    if (typeof string === "string") {
        res = string.split(delimiter)
        if (arguments.length > 2) {
            let limit = arguments[2]
            if (limit < 0) {
                res = res.slice(0, limit)
            } else if (limit === 0) {
                res = [string]
            } else if (limit < res.length) {
                res = res.slice(0, limit - 1).concat(res.slice(limit - 1).join(delimiter))
            }
        }
    }
    return res;
}

console.log(explode(" ", "piece1 piece2 piece3 piece4 piece5 piece6"))
console.log(explode(":", "foo:*:1023:1000::/home/foo:/bin/sh"))
console.log(explode("|", "one|two|three|four", 2))
console.log(explode("|", "one|two|three|four", -1))
console.log(explode("|", "one|two|three|four", 0))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Explode.java
package com.onevgo.functions;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class Explode {
    public static List<String> explode(String delimiter, String str) {
        return explode(delimiter, str, 0);
    }

    public static List<String> explode(String delimiter, String str, int limit) {
        if (str == null) return Collections.emptyList();
        return Arrays.asList(str.split(delimiter, limit));
    }

    public static void main(String[] args) {
        String pizza = "piece1 piece2 piece3 piece4 piece5 piece6";
        System.out.println(explode(" ", pizza));

        String data = "foo:*:1023:1000::/home/foo:/bin/sh";
        System.out.println(explode(":", data));

        String str = "one|two|three|four";
        System.out.println(explode("\\|", str, 2));
    }
}
